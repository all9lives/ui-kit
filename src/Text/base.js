import { css } from 'styled-components'
import Unit from '../utils/Unit'
import Size from '../utils/Size'
import Color from '../utils/Color'

const base = css`
  font-size: ${props => Unit.convert(Size[props.size] || props.size)};
  color: ${props => (Color[props.color] || props.color)};
  margin: ${props => Unit.parse(props.margin, (margin) => margin * Size.spacing)};
  align-self: ${props => props.alignSelf};
`

export default base
