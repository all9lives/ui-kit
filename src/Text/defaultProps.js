import Color from '../utils/Color'
import Unit from '../utils/Unit'
import Size from '../utils/Size'

export default {
  size: Unit.convert(Size.regular),
  color: Color.black
}
