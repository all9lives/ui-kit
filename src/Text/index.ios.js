import Text from './index.android'

export default Text
