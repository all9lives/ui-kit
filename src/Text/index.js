import React from 'react';
import { Text as PrimitiveText } from 'react-primitives';

const Text = ({ children }) => (
  <PrimitiveText>{children}</PrimitiveText>
);

export default Text;
