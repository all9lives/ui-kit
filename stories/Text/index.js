import React from 'react'
import { action } from '@storybook/addon-actions'
import createStory from '../../storybook/createStory'
import Text from '../../src/Text'
import View from '../../src/View'

const squareStyle = { width: 200, height: 200, backgroundColor: 'white' }

createStory('Text', module)
  .add('with Apple', () => <Text>Apple</Text>)
  .add('with Banana', () => <Text>Banana</Text>)
  .add('with size `tiny`', () => <Text size='tiny'>Tiny</Text>)
  .add('with size `large`', () => <Text size='large'>Large</Text>)
  .add('with color `white`', () => <Text color='white'>White</Text>)
  .add('with color `primary`', () => <Text color='primary'>Primary</Text>)
  .add('with margin `tiny`', () => (
    <View>
      <Text margin='tiny'>Margin Tiny</Text>
      <Text margin='tiny'>Margin Tiny</Text>
    </View>
  ))
  .add('with margin `large`', () => (
    <View>
      <Text margin='large'>Margin Large</Text>
      <Text margin='large'>Margin Large</Text>
    </View>
  ))
  .add('with alignSelf `center`', () => (
    <View style={squareStyle}>
      <Text alignSelf='center'>Align Center</Text>
    </View>
  ))
  .add('with onPress', () => (
    <Text onPress={action('Pressed')}>Press Me</Text>
  ))
