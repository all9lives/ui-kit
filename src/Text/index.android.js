import styled from 'styled-components/native'
import base from './base'
import defaultProps from './defaultProps'
import Size from '../utils/Size'

const getLineHeight = (props) => {
  if (props.content) {
    return Size.spacing * 2.5
  }
  if (props.size === 'title') {
    return Size.spacing * 3
  }
  return Size.spacing * 2
}

const Text = styled.Text`
  ${base}
  line-height: ${props => getLineHeight(props)};
`
Text.defaultProps = defaultProps

export default Text
