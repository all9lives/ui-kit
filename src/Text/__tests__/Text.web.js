/* eslint-env jest */
import 'jest-styled-components'
import React from 'react'
import { shallow } from 'enzyme'
import Text from '../index.js'

it('renders a snapshot for `Text` on `web`', () => {
  const wrapper = shallow(
    <Text>Apple</Text>
  )
  expect(wrapper).toMatchSnapshot()
})

it('renders a snapshot for `Text` with `size:tiny` on `web`', () => {
  const wrapper = shallow(
    <Text size='tiny'>Apple</Text>
  )
  expect(wrapper).toMatchSnapshot()
})

it('renders a snapshot for `Text` with `color:blue` on `web`', () => {
  const wrapper = shallow(
    <Text color='blue'>Apple</Text>
  )
  expect(wrapper).toMatchSnapshot()
})
