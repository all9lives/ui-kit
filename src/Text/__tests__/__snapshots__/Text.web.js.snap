// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text.web renders a snapshot 1`] = `
.c0 {
  font-size: 16px;
  color: #282828;
  margin: 0px;
}

<span
  className="c0"
  color="#282828"
  size={16}
>
  Apple
</span>
`;

exports[`Text.web renders a snapshot with \`color:blue\` 1`] = `
.c0 {
  font-size: 16px;
  color: blue;
  margin: 0px;
}

<span
  className="c0"
  color="blue"
  size={16}
>
  Apple
</span>
`;

exports[`Text.web renders a snapshot with \`margin:tiny\` 1`] = `
.c0 {
  font-size: 16px;
  color: #282828;
  margin: 6px;
}

<span
  className="c0"
  color="#282828"
  size={16}
>
  Apple
</span>
`;

exports[`Text.web renders a snapshot with \`size:tiny\` 1`] = `
.c0 {
  font-size: 8px;
  color: #282828;
  margin: 0px;
}

<span
  className="c0"
  color="#282828"
  size="tiny"
>
  Apple
</span>
`;
